dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/internal/ContextProvider.kt

```diff
@@ -9,5 +9,9 @@ package com.datadog.android.v2.core.internal
 import com.datadog.android.v2.api.context.DatadogContext
 
 internal interface ContextProvider {
-    val context: DatadogContext?
+    // TODO RUMM-0000 getting context may be quite heavy, should it be something non-blocking here?
```

Comment:
You mean like an async call ? Maybe we can provide both ?

Reply:
We are currently not sure if we will expose core context provider as a part of public API. For internal needs we are ok with blocking call for now, considering the way we get context fields and what fields are added to the context, so keeping things simple for now.